dapps/evancore.vue.libs/src/dispatcher.ts

```diff
@@ -0,0 +1,245 @@
+        }));
+      } catch (ex) {
+        console.error(ex.message);
+      }
+    }));
+
+    // start dispatcher watching
+    this.dispatcherWatcher = Dispatcher.watch(
+      ($event: any) => this.handleDispatcherEvent($event.detail.dispatcher, $event.detail.instance),
+    );
+  }
+
+  /**
+   * Remove the dispatcher watcher.
+   */
+  destroy() {
+    this.dispatcherWatcher && this.dispatcherWatcher();
+  }
+
+  /**
+   * Load dispatcher from an ens address or return thee cached one.
+   */
+  async loadDispatcher(dispatcherId: string) {
+    const [ dappEns, dispatcherName ] = dispatcherId.split('|||');
+    // load dependencies and dapp content
+    await dappBrowser.dapp.loadDAppDependencies(dappEns, false);
+    const dapp = await dappBrowser.System.import(`${ dappEns }!dapp-content`);
+    const dispatcher = dapp[dispatcherName];
+
+    // add translation to correctly display instance dispatcher titles
+    if (dapp.translations) {
+      Object
+        .keys(dapp.translations)
+        .forEach(key => this.vue.$i18n.add(key, dapp.translations[key]));
+    }
+
+    // push dispatcher id into the list of already loaded, so we only need to load translations once
+    if (this.loadedDispatchers.indexOf(dispatcherId) === -1) {
+      this.loadedDispatchers.push(dispatcherId);
+    }
+
+    return dispatcher;
+  }
+
+  /**
+   * Handle dispatcher instance updates and write them to instance
+   */
+  handleDispatcherEvent(
+    dispatcher: Dispatcher,
+    instance: DispatcherInstance,
+    scope: EvanVueDispatcherHandler = this
+  ) {
+    const [ dappEns, dispatcherName ] = dispatcher.id.split('|||');
+    const isCurr = dappEns === this.dappName;
+    const id = scope === this ? dispatcher.id : dispatcherName
+
+    // fill empty instances
+    scope.instances[id] = scope.instances[id] || [ ];
+
+    // search for existing dispatcher instances
+    const found = scope.instances[id]
+      .find(dataEntry => dataEntry.id === instance.id);
+    const foundIndex = found ? scope.instances[id].indexOf(found) : -1;
+
+    // if instance has finished it's work, clear it
+    if (instance.status === 'finished' || instance.status === 'deleted') {
+      found && scope.instances[id].splice(foundIndex, 1);
+    } else {
+      // update the instances
+      if (foundIndex === -1) {
+        scope.instances[id].push(instance);
+      } else {
+        scope.instances[id][foundIndex] = instance;
+      }
+    }
+
+    // clear old instances from scope (this, curr)
+    if (scope.instances[id].length === 0) {
+      Vue.set(scope.data, id, [ ]);
+      Vue.set(scope.error, id, false);
+      Vue.set(scope.running, id, false);
+    } else {
+      Vue.set(scope.running, id, true);
+      Vue.set(
+        scope.error,
+        id,
+        scope.instances[id].filter((filterVar) => filterVar.status === 'error').length !== 0
+      );
+      Vue.set(scope.data, id, scope.instances[id].map(filterVar => filterVar.data));
```

Comment:
on this line filterVar is bit confusing, would name it `instance` or something like this